pkg/registry/core/shoot/strategy.go

```diff
@@ -67,6 +73,25 @@ func (shootStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Obje
 	if mustIncreaseGeneration(oldShoot, newShoot) {
 		newShoot.Generation = oldShoot.Generation + 1
 	}
+
```

Comment:
Nit: Maybe it's worth to mention that we need to do this after `mustIncreaseGeneration` because otherwise we'd trigger a reconciliation outside the maintenance window.